internal/model/labels.go

```diff
@@ -59,6 +61,53 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	for key, localV := range l {
+		v, ok := labels[key]
+		if !ok {
+			return false
+		}
+		if v.Value != localV.Value {
+			return false
+		}
+		if len(v.Values) != len(localV.Values) {
+			return false
+		}
+		for _, s := range [][]string{v.Values, localV.Values} {
+			if !sort.StringsAreSorted(s) {
+				sort.Strings(s)
+			}
+		}
+		for i, v := range v.Values {
+			if v != localV.Values[i] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// Flatten returns a string interpretation of the labels.
+func (l Labels) Flatten() string {
+	if len(l) == 0 {
+		return ""
+	}
+
+	var strbld strings.Builder
+	for key, value := range l {
+		strbld.WriteString(key)
+		if value.Value != "" {
+			strbld.WriteString(value.Value)
+			continue
+		}
+		for _, v := range value.Values {
```

Comment:
> in cases where both are set, the `Values` field will be used and `Value` will be ignored.

As per the struct docs, `Values` is preferred over `Value`, shouldn't `Flatten` keep the same priority?